torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -150,26 +160,38 @@ class NCCLComm {
   }
 
   ncclComm_t getNcclComm() {
-    std::unique_lock<std::mutex> lock(mutex_);
```

Comment:
Why did we remove the lock?

Reply:
Oops, must've been an accident, not sure how that got removed